### PR TITLE
Remove the warning about the Integration Test plugin

### LINKED ISSUE #71594: Running a test with package:integration_test should supress warnings about missing plugin for package:integration_test
When we run from `flutter drive`, the user sees a message like this:

> Warning: integration_test test plugin was not detected.

This is confusing for users. We should avoid trying to communicate on the channel in that case so the message wn't show up.

/cc @csells @johnpryan

### PR DESCRIPTION
**Pull request by jiahaog**
Fix #71594

Use a compile time constant to skip reporting results to native when running Integration Tests with the Flutter Tool.

### CODE REVIEW

File: packages/integration_test/lib/integration_test.dart [hunk 1]
```diff
@@ -20,24 +20,35 @@ import 'common.dart';
...
+  defaultValue: true,
+);
+
 /// A subclass of [LiveTestWidgetsFlutterBinding] that reports tests results
 /// on a channel to adapt them to native instrumentation test format.
 class IntegrationTestWidgetsFlutterBinding extends LiveTestWidgetsFlutterBinding implements IntegrationTestResults {
   /// Sets up a listener to report that the tests are finished when everything is
   /// torn down.
   IntegrationTestWidgetsFlutterBinding() {
-    // TODO(jackson): Report test results as they arrive
```
Comment (dnfield) [inline packages/integration_test/lib/integration_test.dart:29]: Did you eman to remove this?

File: packages/integration_test/lib/integration_test.dart [hunk 2]
```diff
@@ -52,13 +63,8 @@ class IntegrationTestWidgetsFlutterBinding extends LiveTestWidgetsFlutterBinding
       } on MissingPluginException {
         print('Warning: integration_test test plugin was not detected.');
```
[the remark this answers is not in the document]
Comment (jiahaog) [reply packages/integration_test/lib/integration_test.dart]: It's still needed for those who are using platform native reporting to notify them. Ultimately throwing an exception here does nothing apart from making a stack trace appear in the logs, but the exit code will never be affected should this be reached (unless running with the Flutter Tool)

Including more instructions here is a good idea though, updated the warning. Do you think we should still rethrow here? For those who are currently using platform native plugins, showing them the nasty looking stack trace should be fine, but I'm not sure if we should do the same for those who are running the tests with `flutter drive` today – there's no way to distinguish between the two. What do you think?

File: packages/integration_test/lib/integration_test.dart (hunk 1 above)
Comment (jiahaog) [reply packages/integration_test/lib/integration_test.dart:29]: Yes, I think this is obsolete now. We're already doing so when running with `flutter test`, and I don't think it's on our radar to do so for platform native reporting.

**Review by jonahwilliams — APPROVED** [review]
LGTM

**Comment by jonahwilliams** [comment]
(And if LG to @dnfield )

File: packages/integration_test/lib/integration_test.dart [hunk 3]
```diff
@@ -50,15 +63,22 @@ class IntegrationTestWidgetsFlutterBinding extends LiveTestWidgetsFlutterBinding
           },
         );
       } on MissingPluginException {
-        print('Warning: integration_test test plugin was not detected.');
-      }
-      if (!_allTestsPassed.isCompleted) {
-        _allTestsPassed.complete(true);
+        print(r"""
+Warning: integration_test plugin was not detected.
```
Comment (dnfield) [inline packages/integration_test/lib/integration_test.dart:67]: nit: use `'''`.

**Review by dnfield — APPROVED** [review]
LGTM with nit

**Pull request merged**